src/System.Web.OData/OData/Routing/DefaultODataPathHandler.cs

```diff
@@ -112,12 +112,10 @@ private ODataPath Parse(string serviceRoot, string odataPath, IServiceProvider r
             {
                 Contract.Assert(serviceRoot != null);
 
-                serviceRootUri = new Uri(
-                    serviceRoot.EndsWith("/", StringComparison.Ordinal)
-                        ? serviceRoot
-                        : serviceRoot + "/");
+                string serviceRootStr = serviceRoot.EndsWith("/", StringComparison.Ordinal) ? serviceRoot : serviceRoot + "/";
```

Comment:
I'm unclear how it solves the problem, do you know why?

Reply:
Problem is connected with such creation:
new Uri(new Uri, relativeUri)

I have really simillar problem, but the error i get is "Invalid URI: The Uri scheme is too long." 
It seems that microsoft library System.Uri have some bugs there. For example it takes relative uri and try to find scheme there. So if my OData request have any(a:(...)) with not encoded colon, it takes this colon as end of scheme. And there is limitation to scheme to 1024 signs. 

At least this is my case. I don't know is this the same case, but solution that you gave will probably solve my problem too.

Reply:
@krzysztofwierzbicki Yes, my solution now is to write a new class that inherits the IODataPathHandler interface and instantiate my new class when calling the MapODataServiceRoute method in the WebApiConfig file.

Reply:
Its great idea! Thanks! But still i think about it as temporary solution, as the best would be fix it here.

Reply:
Yes, this is only a temporary solution.If can fix it here, that's the best.